lntest/node.go

```diff
@@ -1666,3 +1761,101 @@ func (hn *HarnessNode) receiveTopologyClientStream(
 		}
 	}
 }
+
+// isEqualChannelPolicy checks that the policy matches the expected one.
+func isEqualChannelPolicy(policy, expectedPolicy *lnrpc.RoutingPolicy) bool {
+	if policy.FeeBaseMsat != expectedPolicy.FeeBaseMsat {
+		return false
+	}
+
+	if policy.FeeRateMilliMsat != expectedPolicy.FeeRateMilliMsat {
+		return false
+	}
+
+	if policy.TimeLockDelta != expectedPolicy.TimeLockDelta {
+		return false
+	}
+
+	if policy.MinHtlc != expectedPolicy.MinHtlc {
+		return false
+	}
+
+	if policy.MaxHtlcMsat != expectedPolicy.MaxHtlcMsat {
+		return false
+	}
+
+	if policy.Disabled != expectedPolicy.Disabled {
+		return false
+	}
+
+	return true
+}
+
+// handlePolicyUpdateWatchRequest checks that if the expected policy can be
+// found either in the node's interval state or describe graph response. If
+// found, it will signal the request by closing the event channel. Otherwise it
+// does nothing but returns nil.
+func (hn *HarnessNode) handlePolicyUpdateWatchRequest(req *chanWatchRequest) {
+	op := req.chanPoint
+
+	// Get a list of known policies for this chanPoint+advertisingNode
+	// combination. Start searching in the node state first.
+	policies, ok := hn.policyUpdates[op.String()][req.advertisingNode]
+
+	if !ok {
+		// If it cannot be found in the node state, try searching it
+		// from the node's DescribeGraph.
+		policyMap := hn.getChannelPolicies(req.includeUnannounced)
+		policies, ok = policyMap[op.String()][req.advertisingNode]
+		if !ok {
+			return
+		}
+	}
+
+	// Check if there's a matched policy.
+	for _, policy := range policies {
+		if isEqualChannelPolicy(policy, req.policy) {
+			close(req.eventChan)
+			return
+		}
+	}
+}
+
+// getChannelPolicies queries the channel graph and formats the policies into
+// the format defined in type policyUpdateMap.
+func (hn *HarnessNode) getChannelPolicies(include bool) policyUpdateMap {
+
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), DefaultTimeout,
+	)
+	defer cancel()
+
+	graph, err := hn.DescribeGraph(ctxt, &lnrpc.ChannelGraphRequest{
+		IncludeUnannounced: include,
+	})
+	if err != nil {
+		hn.AddToLog("DescribeGraph got err: %v", err)
```

Comment:
Ok the more I change the more I feel that, maybe we should use `hn.AddToLog` to print to console instead of the log?

Reply:
I was thinking the same... Because downloading the logs from Travis is a bit tedious anyway. Or we don't log but indeed start to panic in cases like this. That should make it very obvious why the itests fail...